src/Library/demos/Grid/main.js

```diff
@@ -0,0 +1,53 @@
+const button_ids = [
+  "button00",
+  "button01",
+  "button02",
+  "button10",
+  "button11",
+  "button12",
+  "button20",
+  "button21",
+  "button22",
+];
+
+for (const id of button_ids) {
+  const button = workbench.builder.get_object(id);
+  button.connect("clicked", onClicked);
+}
+
+let step = 1;
+function onClicked(button) {
+  //check access for user action
+  let clicked = button.get_child();
+  if (
+    clicked.icon_name !== "cross-large-symbolic" &&
+    clicked.icon_name !== "circle-outline-thick-symbolic"
+  ) {
```

Comment:
an early return would make the function easier to read and reduce one level of indentation

something like

```suggestion
  if (clicked.icon_name) return;
```